_posts/Durr_Hoyer_Post_1.md

```diff
@@ -0,0 +1,126 @@
+---
+title: "A quantum algorithm for finding the minimum of an unsorted table"
+date:
+categories:
+  - Blog
+tags:
+  - qsharp
+  - algorithm
+  - open source
+---
+### Durr Hoyer
+
+Hello all! My name is Mridul Sarkar. I am an undergraduate at University of California Davis studying Scientific and Mathematical Computation. I began my journey from math to computers through my passion for ML algorithms. Eventually I stumbled upon quantum computing. I was instantly pulled in by the elegant algorithms and proposed efficiency.
+
+
+
+### Background
+Recently I began working on quantum computing with an idea to treat nodes of neural networks as qubits. From here I took a deep dive into the research papers and found myself amazed by the research being produced. I began with a Coursera course from St. Petersburg University on quantum computing with detail on quantum algorithm design and quantum computer architecture. From here I tinkered with Q# and developed basic algorithms that I learnt. I found Durr and Hoyer's paper and saw it as an ensemble of Duestchs, Shors, and Grovers Algorithm; a perfect stepping stone coming from my background. This project has pushed me and is still pushing me outside of my comfort zone. Learning Q# while slowly refining my quantum mechanics has been a rewarding experience. I would like to pass on what I have seen and understood in that time while open to feedback. I will likely be posting about this algorithm more as it continues developing and I learn more.
+
+
+### Motivation
+The motivation behind this project is to provide open source functionality. The efficiency that is proposed by this algorithm is Big-O = square root of the number of items in the table. Durr and Hoyer propose an ensemble of quantum algorithms in order to find the minimum of an unsorted table with a 50% success rate. This success rate is bounded by the Big-O. If we decide to alleviate this Big-O the probability of finding the minimum increases. With this comes some complexity in the inner workings of the library which will be explained when analyzing the algorithm. In order for this library to be used properly it must meet some guidelines. At the moment I am referencing the amazing template given by Dr. Sarah Kaiser https://github.com/crazy4pi314/qsharp-library-template in order to make this library usable in Q#. I am programming the algorithm in Q# with a python shell. The python shell will be provided below. The circuit's gates will be analyzed and some code is given.
+
+
+### The Algorithm
+A brief introduction to the Durr and Hoyer Algorithm can be derived from the Quantum Minimum Searching Algorithm outlined in 'A quantum algorithm for finding the minimum' [(1)]:
+
+Durr and Hoyer propose a computationally efficient routine to find the minimum of a table of distinct unsorted integers. The algorithm is as follows:
+
+
+![QMSA](/assets/images//DurrHoyer-QMSA.JPG "QMSA")
+
+
+#### Psuedocode
```

Comment:
Can you maybe describe the steps in words instead? It's not super accessible for people using screen readers to have some of the most important info in the post. You don't have to worry about the Python then.